iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -152,14 +155,29 @@ void ServiceRegistry::find(ServiceDescriptionVector_t& searchResult,
         auto range = m_serviceMap.equal_range(service.value());
         for (auto entry = range.first; entry != range.second; ++entry)
         {
-            searchResult.push_back(m_serviceDescriptionVector[entry->second]);
+            serviceInstanceResult.push_back(m_serviceDescriptionVector[entry->second]);
         }
     }
     else
     {
         // Find (*, *)
         // O(n)
-        searchResult = m_serviceDescriptionVector;
+        serviceInstanceResult = m_serviceDescriptionVector;
+    }
+
+    if (event)
+    {
+        for (auto iterator = serviceInstanceResult.begin(); iterator != serviceInstanceResult.end(); iterator++)
+        {
+            if (iterator->serviceDescription.getEventIDString() == event.value())
+            {
+                searchResult.push_back(*iterator);
+            }
+        }
+    }
+    else
+    {
+        searchResult = serviceInstanceResult;
```

Comment:
This logically correct but costly (as we cannot move the result). We will probably change it soon anyway. Ideally we would fill the search result once in the way it is needed (in this branch the `serviceInstanceResult` is already the search result.

As it is one way would probably need to check earlier (e.g. in each case) whether we have an `event` wildcard.

But this would make the code less concise and it is not optimized for efficiency now anyway. I do not see what can be done to remove this inefficiency in a simple way.

Reply:
As I stated in the `Notes for Reviewer`, I know it's quite inefficient. Since we probably will change the underlying data structure soon, is it okay for you to leave it as it is @MatthiasKillat ?

Reply:
Sure. And I missed the notes somehow.